src/main/java/io/temporal/samples/hello/HelloSideEffect.java

```diff
@@ -0,0 +1,266 @@
+ * only use the SideEffects return value!
+ *
+ * <p>To execute this example a locally running Temporal service instance is required. You can
+ * follow instructions on how to set up your Temporal service here:
+ * https://github.com/temporalio/temporal/blob/master/README.md#download-and-start-temporal-server-locally
+ */
+public class HelloSideEffect {
+
+  // Define the task queue name
+  static final String TASK_QUEUE = "HelloSideEffectTaskQueue";
+
+  // Define our workflow unique id
+  static final String WORKFLOW_ID = "HelloSideEffectTaskWorkflow";
+
+  /**
+   * Define the Workflow Interface. It must contain one method annotated with @WorkflowMethod.
+   *
+   * <p>Workflow code includes core processing logic. It that shouldn't contain any heavyweight
+   * computations, non-deterministic code, network calls, database operations, etc. All those things
+   * should be handled by Activities.
+   *
+   * @see io.temporal.workflow.WorkflowInterface
+   * @see io.temporal.workflow.WorkflowMethod
+   */
+  @WorkflowInterface
+  public interface SideEffectWorkflow {
+
+    /**
+     * This method is executed when the workflow is started. The workflow completes when the
+     * workflow method finishes execution.
+     */
+    @WorkflowMethod
+    String start();
+  }
+
+  /**
+   * Define the Activity Interface. Activities are building blocks of any temporal workflow and
+   * contain any business logic that could perform long running computation, network calls, etc.
+   *
+   * <p>Annotating activity methods with @ActivityMethod is optional
+   *
+   * @see io.temporal.activity.ActivityInterface
+   * @see io.temporal.activity.ActivityMethod
+   */
+  @ActivityInterface
+  public interface SideEffectActivities {
+
+    // Define your activity method which can be called during workflow execution
+    String doSomething(String input);
+  }
+
+  // Define the workflow implementation which implements our execute workflow method.
+  public static class SideEffectWorkflowImpl implements SideEffectWorkflow {
+
+    /**
+     * Define the SideEffectActivities stub. Activity stubs are proxies for activity invocations
+     * that are executed outside of the workflow thread on the activity worker, that can be on a
+     * different host. Temporal is going to dispatch the activity results back to the workflow and
+     * unblock the stub as soon as activity is completed on the activity worker.
+     *
+     * <p>Let's take a look at each {@link ActivityOptions} defined: The "setScheduleToCloseTimeout"
+     * option sets the overall timeout that our workflow is willing to wait for activity to
+     * complete. For this example it is set to 2 seconds.
+     */
+    private final SideEffectActivities activities =
+        Workflow.newActivityStub(
+            SideEffectActivities.class,
+            ActivityOptions.newBuilder().setScheduleToCloseTimeout(Duration.ofSeconds(2)).build());
```

Comment:
Never use ScheduleToClose timeout in the samples. use StartToClose instead.

Reply:
lots of samples use setScheduleToCloseTimeout. will update here and also update across examples in separate pr

Reply:
I recently fixed that in Go samples: https://github.com/temporalio/samples-go/pull/89

Reply:
https://github.com/temporalio/samples-java/pull/96

note that i had to add retries to fileprocessingworkflow there in order to get the tests to passs. I am still trying to figure out why this is needed when the timeout changes to setStartToCloseTimeout in that case